drivers/scheduler/spec/factory.go

```diff
@@ -49,12 +44,7 @@ func (f *Factory) GetAll() []*AppSpec {
 	var specs []*AppSpec
 	for _, val := range appSpecFactory {
 		if val.Enabled {
-			valCopy, err := scheme.Scheme.DeepCopy(val)
-			if err != nil {
-				logrus.Errorf("Failed to create a copy of spec object: %#v Err: %v", val, err)
-				return nil
-			}
-
+			valCopy := deepcopy.Copy(val)
```

Comment:
may be have a nil check before we type cast it to avoid panic?
and not append if it is nil.
Same for Get, return nil

Reply:
Added a type assertion, please review the update